lib/plugins/filter/after_post_render/external_link.js

```diff
@@ -1,15 +1,23 @@
 'use strict';
 
-const { parse } = require('url');
+const { URL } = require('url');
+
+const urlObj = (str) => {
+  try {
+    return new URL(str);
+  } catch (err) {
+    return str;
+  }
+};
 
 const isExternal = (url, config) => {
   const exclude = Array.isArray(config.external_link.exclude) ? config.external_link.exclude :
     [config.external_link.exclude];
-  const data = parse(url);
+  const data = urlObj(url);
   const host = data.hostname;
-  const sitehost = parse(config.url).hostname || config.url;
+  const sitehost = urlObj(config.url).hostname || config.url;
 
-  if (!data.protocol || !sitehost) return false;
+  if (!data.protocol || !sitehost || !data.origin) return false;
```

Comment:
when there is no data.origin, it returns a string value of `'null'`, not null type.

`!data.protocol` should be removed, because it's not null even in `javascript:foobar`.

``` js
const url = new URL('javascript:foobar')
console.log(url.protocol)
// 'javascript:'
```

Reply:
`data` variable will be string type if `url` is internal link.

you need to check the type of `data` first, otherwise `data.origin` will cause error if internal link.

``` js
if (!sitehost || typeof data === 'string') return false;
if (data.origin === 'null') return false;
```